src/devices/cpu/t11/t11ops.hxx

```diff
@@ -469,14 +536,14 @@ void t11_device::asl_ded(uint16_t op)       { m_icount -= 30; { ASL_M(DED); } }
 void t11_device::asl_ix(uint16_t op)        { m_icount -= 30; { ASL_M(IX);  } }
 void t11_device::asl_ixd(uint16_t op)       { m_icount -= 36; { ASL_M(IXD); } }
 
-void t11_device::sxt_rg(uint16_t op)        { m_icount -= 12; { SXT_R(RG);  } }
-void t11_device::sxt_rgd(uint16_t op)       { m_icount -= 21; { SXT_M(RGD); } }
-void t11_device::sxt_in(uint16_t op)        { m_icount -= 21; { SXT_M(IN);  } }
-void t11_device::sxt_ind(uint16_t op)       { m_icount -= 27; { SXT_M(IND); } }
-void t11_device::sxt_de(uint16_t op)        { m_icount -= 24; { SXT_M(DE);  } }
-void t11_device::sxt_ded(uint16_t op)       { m_icount -= 30; { SXT_M(DED); } }
-void t11_device::sxt_ix(uint16_t op)        { m_icount -= 30; { SXT_M(IX);  } }
-void t11_device::sxt_ixd(uint16_t op)       { m_icount -= 36; { SXT_M(IXD); } }
+void t11_device::sxt_rg(uint16_t op)        { CHECK_IS(IS_LEIS); m_icount -= 12; { SXT_R(RG);  } }
+void t11_device::sxt_rgd(uint16_t op)       { CHECK_IS(IS_LEIS); m_icount -= 21; { SXT_M(RGD); } }
+void t11_device::sxt_in(uint16_t op)        { CHECK_IS(IS_LEIS); m_icount -= 21; { SXT_M(IN);  } }
+void t11_device::sxt_ind(uint16_t op)       { CHECK_IS(IS_LEIS); m_icount -= 27; { SXT_M(IND); } }
+void t11_device::sxt_de(uint16_t op)        { CHECK_IS(IS_LEIS); m_icount -= 24; { SXT_M(DE);  } }
+void t11_device::sxt_ded(uint16_t op)       { CHECK_IS(IS_LEIS); m_icount -= 30; { SXT_M(DED); } }
+void t11_device::sxt_ix(uint16_t op)        { CHECK_IS(IS_LEIS); m_icount -= 30; { SXT_M(IX);  } }
+void t11_device::sxt_ixd(uint16_t op)       { CHECK_IS(IS_LEIS); m_icount -= 36; { SXT_M(IXD); } }
```

Comment:
Runtime checks are going to be expensive – is there a better way?

Reply:
If you're implying subclassing might be the solution, that may not work well here, since the PDP-11 family architecture left no clear main line of descent and the extensions to the original 11/20 instruction set were supported on a notoriously haphazard basis. Occasionally an instruction class that had built-in or optional support on one processor model would not be supported at all on a model nominally succeeding it.

Reply:
I was more suggesting either using a static dispatch table of some kind (e.g. an array of function points or something), or using generated dispatch functions to deal with instruction set differences between variants.

Reply:
I am unsure how to best handle multiple variants in one core, this as a first draft.   FWIW, SIMH uses feature flags and a giant switch() statement.